akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -171,7 +173,15 @@ import akka.stream.scaladsl.Source
         Source
           .futureSource(handler.tryStart().flatMap(_ => sourceProvider.source(readOffsets)))
           .via(killSwitch.flow)
-          .map(envelope => sourceProvider.extractOffset(envelope) -> envelope)
+          .mapConcat { env =>
+            val offset = sourceProvider.extractOffset(env)
+            sourceProvider.verifyOffset(offset) match {
+              case Success => List(offset -> env)
+              case SkipOffset(reason) =>
+                logger.warning("Source provider instructed projection to skip record with reason: {}", reason)
+                Nil
+            }
```

Comment:
I think it could be more efficient to use `filter` than `mapConcat` here

Reply:
:+1: